### PR TITLE
FileCache: Increase CHUNK_SIZE to 128 KB

### PR DESCRIPTION
**Pull request by fritsch**
I need that to play highbitrate videos over my wireless bridge.

### CODE REVIEW

**Comment by fritsch** [comment]
@popcornmix @Memphiz @FernetMenta as we talked before about that. Okay?

**Comment by popcornmix** [comment]
Does it need to be that high? We've been testing 128K in milhouse builds.
We know 1M causes very slow opening of ISO files (from below 10s to over 30s)
http://forums.kodi.tv/showthread.php?tid=250817&pid=2307214#pid2307214
so I'm a bit nervous about 512K.

**Comment by fritsch** [comment]
Can't we let this user test a build with 512 and one with 128? And ask which one is better? (Without telling in advance?)

**Comment by fritsch** [comment]
Btw. I chose this 512 cause I have seen it in a @memphiz thread.  I am happy with 128, too.

**Comment by popcornmix** [comment]
I've pinged the original guy from forum.
Memphiz did say that libnfs should use the chunksize the server reports:
http://forum.kodi.tv/showthread.php?tid=298461&pid=2503108&highlight=memphiz#pid2503108
Can you confirm if that works for you?

**Comment by fritsch** [comment]
I think you asked me to test that the last time, basically: https://github.com/fritsch/xbmc/blob/cc8ee867d472cbe7991fb06452b49c77e39006b1/xbmc/filesystem/FileCache.cpp#L177 would end up with 64 KB as NFS returns just 1.

**Comment by fritsch** [comment]
Btw. this would be:
```
20:04:06.593 T:140226060084992   DEBUG: NFS: chunks: r/w 1048576/1048576
```

1 MB.

**Comment by fritsch** [comment]
Btw. currently the write is clamped to 32 KB also - with that change I have a bad feeling :-)

**Comment by arnova** [comment]
I'm wondering: Why does a bigger chunk-size help here? Is the NFS implementation that crappy? And if it's an NFS-problem, why increase it for all protocols?

**Comment by MrMC** [comment]
FYI: https://lists.samba.org/archive/samba-technical/2014-February/097855.html

**Comment by fritsch** [comment]
@arnova requesting small chunks in a row over weak connection with varying
response time is really bad in general no matter what protocol.

Am 07.02.2017 2:43 nachm. schrieb "Scott Davilla" <notifications@github.com

**Review by Memphiz — COMMENTED** [review]
I think this change might be correct regardless of the cache size...

**Comment by fritsch** [comment]
Rebased and added a commit by @popcornmix

**Comment by popcornmix** [comment]
These commits were in last Milhouse build.
Previously I could only play live HD TV recordings over wifi with buffermode=1.
Now I can play them without that setting.
So +1 from me.

**Comment by fritsch** [comment]
@FernetMenta is the ffmpeg change fine with you?

**Comment by MrMC** [comment]
uint8_t probe_buffer[FFMPEG_FILE_BUFFER_SIZE + AVPROBE_PADDING_SIZE];

be careful you don't blow up your local stack.

**Comment by fritsch** [comment]
@MrMC okay with a smartptr?

**Comment by MrMC** [comment]
smartptr is ok

**Comment by fritsch** [comment]
jenkins build and merge

(got fernet's OK on IRC).

File: xbmc/cores/VideoPlayer/DVDDemuxers/DVDDemuxFFmpeg.cpp [hunk 1]
```diff
@@ -295,10 +295,10 @@ bool CDVDDemuxFFmpeg::Open(CDVDInputStream* pInput, bool streaminfo, bool filein
       if (trySPDIFonly || (iformat && strcmp(iformat->name, "wav") == 0))
       {
         AVProbeData pd;
-        uint8_t probe_buffer[FFMPEG_FILE_BUFFER_SIZE + AVPROBE_PADDING_SIZE];
+        std::unique_ptr<uint8_t[]> probe_buffer (new uint8_t[FFMPEG_FILE_BUFFER_SIZE + AVPROBE_PADDING_SIZE]);
```
Comment (FernetMenta) [inline xbmc/cores/VideoPlayer/DVDDemuxers/DVDDemuxFFmpeg.cpp]: I don't think a smart pointer is a good idea for a buffer handed to ffmpeg. In this case the variable goes out of scope while ffmped still has a ref to the buffer

Comment (popcornmix) [inline xbmc/cores/VideoPlayer/DVDDemuxers/DVDDemuxFFmpeg.cpp]: But is this any different to previous code? There, a buffer on stack was passed in, which will disappear when scope is left.

**Comment by fritsch** [comment]
Yeah.

Am 26.04.2017 2:50 nachm. schrieb "popcornmix" <notifications@github.com>:

**Comment by FernetMenta** [comment]
I am not sure. local variable are in general allocated at the start of a function.

**Comment by popcornmix** [comment]
I'm pretty sure C/C++ only guarantees allocation within the closest scope of a block.
If it is allocated at start of function that is purely a compiler choice (that may change with versions/optimisation level).
See: http://stackoverflow.com/questions/30702717/memory-allocation-of-local-variables-within-nested

Possibly the code has always been broken and we are only now getting hit.
When would be best to allocate and free?
Make probe_buffer a member variable, allocate in the same local scope and free in destructor?

**Pull request merged**